### DISCUSSION TITLE
FilePath

### DISCUSSION THREAD

**Discussion by dadsoldaxe** [GoyaPtyLtd/BaseElements-Plugin#227] [Q&A]
I am trying to use the "BE_ExportFieldContents" function to export a file from a container field on the server, but I am not able to get it working. I have a feeling it is the file path that is causing an issue, and I have tried many options, but I am not really sure what to do.

If I use a "Save record at PDF" script step it is able to land in the correct directory. I read an article saying that plugins use a different path, but it did not have an example, so I am unsure how to modify the path to suit the plugin.

This is the path I am using for the "Save Record as PDF":
Get ( DocumentsPath ) & "OutPutPDF/filename.pdf

Here is an example of one of the iterations of the path I am using for the BE function:
"/Volumes" & Get ( DocumentsPath ) & "OutPutPDF/" & GetContainerAttribute ( TestingTable::PDF ; "filename" )

When I compare the output from the above with the terminal they are identical.

I am using FileMaker Pro on a Mac and the server is also a Mac running FileMaker server 21.0.1.51, on an M2 running MacOS 14.5.

Thanks.

**Comment by nickorr** [comment]
There's an FAQ about this here : 

[https://github.com/GoyaPtyLtd/BaseElements-Plugin/blob/master/docs/FAQ.md#notes-about-file-paths](https://github.com/GoyaPtyLtd/BaseElements-Plugin/blob/master/docs/FAQ.md#notes-about-file-paths)

In short, use the built in conversion function [ConvertFromFileMakerPath](https://help.claris.com/en/pro-help/content/convert-from-filemaker-path.html) to convert the path from FMP to plugin ( and vice versa ) whenever you use a Get function path.

Cheers,
Nick

**Reply by dadsoldaxe** [reply]
Yep, thanks Nick everything works well running on FileMaker Pro, but when I run it on the Server it returns "?" and does not export the file.

**Reply by nickorr** [reply]
Are you using the format of the server in the function?  If you code it on a Mac, but run it on FMS on windows, the **format** in the convert function needs to adapt to the OS of the device it runs on.

Other than that, trap for all the errors :

Note the paths, before and after conversion.
Check if the folder you're putting it into exists.
Look at the BE Last Error.

**Reply by dadsoldaxe** [reply]
All the computers (server, client, development) are Apple Silicon Mac's. The code that is being run on the server I have attached as an image.  I have added the "BE Last Error" to the Exit Script so I can see it, but it is returning a "?" as well.

<img width="869" alt="image" src="https://github.com/user-attachments/assets/e44a734a-f8c7-4777-860f-bbd7a068f03a">

**Reply by nickorr** [reply]
If the Last Error function returns ? then the plugin isn't installed or active.  IF you think it should be installed, it may need a restart on the server to appear - there are bugs in the update plugin operations where it fails to unload an old plugin.

**Reply by dadsoldaxe** [reply]
thanks, I will give that a go.

**Reply by nickorr** [reply]
Also consider trapping for a version number of the plugin at the top of the script too, that helps avoid having the rest of the code run, and would pick up the same issue.

**Reply by dadsoldaxe** [reply]
ok, so I tried to return a version number and I didn't get one, and I knew I had the BE plugin on that machine as I was using it a few months ago. So I logged into the Server Admin Console and the BE Plugin was there but it was turned off. Whenever I tried to turn it on it turns off immediately - visible on refresh. 

I thought I would upgrade the server just incase, so I deleted the BE plugin from the extension folder and re-installed the server to upgrade from 21.0.1.51 to 21.0.2.202. Restarted the server and dowloaded BE 5.0.0 and installed it using the script step as per normal and it still will not stay on. I then copied the BE plugin manually and it also would not stay on.

Then I read this ...
https://community.claris.com/en/s/question/0D5Vy00000E3nzjKAB/baseelements-plugin-crashes-when-attempting-to-start-on-filemaker-server-2101-and-macos-14-sonoma

I installed the BE Beta and everything is working ... it was a bit of an excursion, and turns out there was nothing wrong with my first cut of the code. Thanks for your help though.

**Reply by nickorr** [reply]
Sorry, yep should have noticed the Apple Silicon.  This is the most recent Mac build : 

https://goya.com.au/files/beplugin/5.1.0b1/BaseElements.fmplugin.dmg

It is beta, but passes every test we have thrown at it, and doesn't have the image issues that were in the first build I did and reference in that community post.

**Comment by MacEasyComputingInc** [comment]
Many thanks for answering this,  since this was not obvious and fixed my problem immediately.  Not sure how I would've known to do this otherwise
